connectivity/check/context.go

```diff
@@ -528,17 +533,62 @@ func (ct *ConnectivityTest) enableHubbleClient(ctx context.Context) error {
 	return nil
 }
 
-func (ct *ConnectivityTest) validateExternalFromCIDRsWithNodesWithoutCilium() error {
-	if len(ct.params.ExternalFromCIDRs) == 0 {
-		ct.Fatalf("--%s must not be empty if Cilium was install with --%s set", "external-from-cidrs", "nodes-without-cilium")
+func (ct *ConnectivityTest) detectPodCIDRs(ctx context.Context) error {
+	nodes, err := ct.client.ListNodes(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list nodes: %w", err)
 	}
-	for _, cidr := range ct.params.ExternalFromCIDRs {
-		addr, err := netip.ParseAddr(cidr)
-		if err != nil {
-			return fmt.Errorf("unable to parse external-from-cidr %q: %w", cidr, err)
+
+	for _, n := range nodes.Items {
+		for _, cidr := range n.Spec.PodCIDRs {
+			f := GetIPFamily(ct.hostNetNSPodsByNode[n.Name].Pod.Status.HostIP)
+			if strings.Contains(cidr, ":") && f == IPFamilyV4 {
+				// Skip if the host IP of the pod mismatches with pod CIDR.
+				// Cannot create a route with the gateway IP family
+				// mismatching the subnet.
+				continue
+			}
+			hostIP := ct.hostNetNSPodsByNode[n.Name].Pod.Status.HostIP
+			ct.params.PodCIDRs = append(ct.params.PodCIDRs, podCIDRs{cidr, hostIP})
 		}
-		ct.params.ExternalFromCIDRMasks = append(ct.params.ExternalFromCIDRMasks, addr.BitLen())
 	}
+
+	return nil
+}
+
+func (ct *ConnectivityTest) detectNodesWithoutCiliumIPs() error {
+	for _, n := range ct.nodesWithoutCilium {
+		pod := ct.hostNetNSPodsByNode[n]
+		for _, ip := range pod.Pod.Status.PodIPs {
```

Comment:
What if `ct.hostNetNSPodsByNode` doesn't contain `n` key? Probably there will be panic in runtime (coz `Pod` is pointer), isn't it?

Reply:
Yep, it will panic. However, it's not expected that such an entry does not exist, so the panic would happen only if there is a bug in the code.